src/anomalib/pre_processing/pre_processing.py

```diff
@@ -109,63 +99,49 @@ def __init__(
         self.predict_transform = self.test_transform
         self.export_transform = get_exportable_transform(self.test_transform)
 
-    def setup_datamodule_transforms(self, datamodule: "AnomalibDataModule") -> None:
-        """Set up datamodule transforms."""
-        # If PreProcessor has transforms, propagate them to datamodule
-        if any([self.train_transform, self.val_transform, self.test_transform]):
-            transforms = {
-                "fit": self.train_transform,
-                "val": self.val_transform,
-                "test": self.test_transform,
-                "predict": self.predict_transform,
-            }
-
-            for stage, transform in transforms.items():
-                if transform is not None:
-                    set_datamodule_stage_transform(datamodule, transform, stage)
-
-    def setup_dataloader_transforms(self, dataloaders: "EVAL_DATALOADERS | TRAIN_DATALOADERS") -> None:
-        """Set up dataloader transforms."""
-        if isinstance(dataloaders, DataLoader):
-            dataloaders = [dataloaders]
-
-        # If PreProcessor has transforms, propagate them to dataloaders
-        if any([self.train_transform, self.val_transform, self.test_transform]):
-            transforms = {
-                "train": self.train_transform,
-                "val": self.val_transform,
-                "test": self.test_transform,
-            }
-            set_dataloaders_transforms(dataloaders, transforms)
-            return
-
-        # Try to get transforms from dataloaders
-        if dataloaders:
-            dataloaders_transforms = get_dataloaders_transforms(dataloaders)
-            if dataloaders_transforms:
-                self.train_transform = dataloaders_transforms.get("train")
-                self.val_transform = dataloaders_transforms.get("val")
-                self.test_transform = dataloaders_transforms.get("test")
-                self.predict_transform = self.test_transform
-                self.export_transform = get_exportable_transform(self.test_transform)
-
-    def setup(self, trainer: Trainer, pl_module: LightningModule, stage: str) -> None:
-        """Configure transforms at the start of each stage.
-
-        Args:
-            trainer: The Lightning trainer.
-            pl_module: The Lightning module.
-            stage: The stage (e.g., 'fit', 'validate', 'test', 'predict').
-        """
-        stage = TrainerFn(stage).value  # Ensure stage is str
+    def on_train_batch_start(self, trainer: Trainer, pl_module: LightningModule, batch: Batch, batch_idx: int) -> None:
```

Comment:
more like a styling issue.. it might be an idea to use multi-lines here as well, similar to other hooks.